// fb-messenger-bot-api/dist/classes/templates/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AbstractMessageTemplate_1 = require("./AbstractMessageTemplate");
exports.AbstractMessageTemplate = AbstractMessageTemplate_1.AbstractMessageTemplate;
var AirlineBoardingPassTemplate_1 = require("./AirlineBoardingPassTemplate");
exports.AirlineBoardingPassTemplate = AirlineBoardingPassTemplate_1.AirlineBoardingPassTemplate;
var AirlineCheckInTemplate_1 = require("./AirlineCheckInTemplate");
exports.AirlineCheckInTemplate = AirlineCheckInTemplate_1.AirlineCheckInTemplate;
var AirlineFlightUpdateTemplate_1 = require("./AirlineFlightUpdateTemplate");
exports.AirlineFlightUpdateTemplate = AirlineFlightUpdateTemplate_1.AirlineFlightUpdateTemplate;
var AirlineItineraryTemplate_1 = require("./AirlineItineraryTemplate");
exports.AirlineItineraryTemplate = AirlineItineraryTemplate_1.AirlineItineraryTemplate;
var ButtonTemplate_1 = require("./ButtonTemplate");
exports.ButtonTemplate = ButtonTemplate_1.ButtonTemplate;
var GenericTemplate_1 = require("./GenericTemplate");
exports.GenericTemplate = GenericTemplate_1.GenericTemplate;
var ListTemplate_1 = require("./ListTemplate");
exports.ListTemplate = ListTemplate_1.ListTemplate;
var MediaTemplate_1 = require("./MediaTemplate");
exports.MediaTemplate = MediaTemplate_1.MediaTemplate;
var OpenGraphTemplate_1 = require("./OpenGraphTemplate");
exports.OpenGraphTemplate = OpenGraphTemplate_1.OpenGraphTemplate;
var ReceiptTemplate_1 = require("./ReceiptTemplate");
exports.ReceiptTemplate = ReceiptTemplate_1.ReceiptTemplate;
