// fb-messenger-bot-api/dist/util/Utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const request_1 = __importDefault(require("request"));
class Utils {
    /* istanbul ignore next line */
    constructor() { }
    static getRequestOptions() {
        return JSON.parse(JSON.stringify(Utils.requestOptions));
    }
    static sendMessage(options, requestData, cb) {
        options.qs.access_token = requestData.token;
        if (requestData.hasOwnProperty('proxy')) {
            options.proxy = requestData.proxy;
        }
        if (typeof cb !== 'function') {
            return new Promise((resolve, reject) => {
                request_1.default(options, (err, _res, body) => {
                    if (err) {
                        reject(err);
                    }
                    else {
                        if (!((typeof body === 'object' && !Array.isArray(body) && body !== null))) {
                            const bodyObject = JSON.parse(body);
                            if (bodyObject.error) {
                                reject(bodyObject.error.message);
                            }
                            else {
                                resolve(bodyObject);
                            }
                        }
                        else {
                            resolve(body);
                        }
                    }
                });
            });
            // tslint:disable-next-line
        }
        else {
            request_1.default(options, (err, _res, body) => {
                if (err)
                    return cb(err);
                if (body.error)
                    return cb(body.error);
                cb(null, body);
            });
            return Promise.resolve();
        }
    }
    static getProxyData(requestData, proxyData) {
        if (proxyData) {
            if (Object.prototype.toString.call(proxyData) === '[object Object]'
                && proxyData.hasOwnProperty('hostname')
                && proxyData.hasOwnProperty('port')) {
                if (proxyData.hostname.indexOf('http') === 0) {
                    requestData.proxy = `${proxyData.hostname}:${proxyData.port}`;
                }
                else {
                    requestData.proxy = `http://${proxyData.hostname}:${proxyData.port}`;
                }
            }
            else {
                throw new Error('Invalid Proxy object given, expected hostname and port');
            }
        }
        return requestData;
    }
}
exports.Utils = Utils;
Utils.requestOptions = {
    url: 'https://graph.facebook.com/v3.1/',
    qs: {
        access_token: undefined,
    },
    method: undefined,
};
