// fb-messenger-bot-api/dist/enums/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var QUICK_REPLY_TYPE;
(function (QUICK_REPLY_TYPE) {
    QUICK_REPLY_TYPE["TEXT"] = "text";
    QUICK_REPLY_TYPE["LOCATION"] = "location";
    QUICK_REPLY_TYPE["USER_PHONE_NUMBER"] = "user_phone_number";
    QUICK_REPLY_TYPE["USER_EMAIL"] = "user_email";
})(QUICK_REPLY_TYPE = exports.QUICK_REPLY_TYPE || (exports.QUICK_REPLY_TYPE = {}));
var LIST_TOP_ELEMENT_STYLE;
(function (LIST_TOP_ELEMENT_STYLE) {
    LIST_TOP_ELEMENT_STYLE["LARGE"] = "large";
    LIST_TOP_ELEMENT_STYLE["COMPACT"] = "compact";
})(LIST_TOP_ELEMENT_STYLE = exports.LIST_TOP_ELEMENT_STYLE || (exports.LIST_TOP_ELEMENT_STYLE = {}));
var BUTTON_PAYMENT_TYPE;
(function (BUTTON_PAYMENT_TYPE) {
    BUTTON_PAYMENT_TYPE[BUTTON_PAYMENT_TYPE["FIXED_AMOUNT"] = 0] = "FIXED_AMOUNT";
    BUTTON_PAYMENT_TYPE[BUTTON_PAYMENT_TYPE["FLEXIBLE_AMOUNT"] = 1] = "FLEXIBLE_AMOUNT";
})(BUTTON_PAYMENT_TYPE = exports.BUTTON_PAYMENT_TYPE || (exports.BUTTON_PAYMENT_TYPE = {}));
var ATTACHMENT_TYPE;
(function (ATTACHMENT_TYPE) {
    ATTACHMENT_TYPE["IMAGE"] = "image";
    ATTACHMENT_TYPE["AUDIO"] = "audio";
    ATTACHMENT_TYPE["VIDEO"] = "video";
    ATTACHMENT_TYPE["FILE"] = "file";
    ATTACHMENT_TYPE["TEMPLATE"] = "template";
    ATTACHMENT_TYPE["FALLBACK"] = "fallback";
})(ATTACHMENT_TYPE = exports.ATTACHMENT_TYPE || (exports.ATTACHMENT_TYPE = {}));
var MESSAGE_TEMPLATE_TYPE;
(function (MESSAGE_TEMPLATE_TYPE) {
    MESSAGE_TEMPLATE_TYPE["BUTTON"] = "button";
    MESSAGE_TEMPLATE_TYPE["LIST"] = "list";
    MESSAGE_TEMPLATE_TYPE["GENERIC"] = "generic";
    MESSAGE_TEMPLATE_TYPE["MEDIA"] = "media";
    MESSAGE_TEMPLATE_TYPE["OPEN_GRAPH"] = "open_graph";
    MESSAGE_TEMPLATE_TYPE["RECEIPT"] = "receipt";
    MESSAGE_TEMPLATE_TYPE["AIRLINE_BOARDING_PASS"] = "airline_boardingpass";
    MESSAGE_TEMPLATE_TYPE["AIRLINE_CHECKIN"] = "airline_checkin";
    MESSAGE_TEMPLATE_TYPE["AIRLINE_ITINERARY"] = "airline_itinerary";
    MESSAGE_TEMPLATE_TYPE["AIRLINE_UPDATE"] = "airline_update";
})(MESSAGE_TEMPLATE_TYPE = exports.MESSAGE_TEMPLATE_TYPE || (exports.MESSAGE_TEMPLATE_TYPE = {}));
var BUTTON_TYPE;
(function (BUTTON_TYPE) {
    BUTTON_TYPE["BUY"] = "payment";
    BUTTON_TYPE["CALL"] = "phone_number";
    BUTTON_TYPE["GAME_PLAY"] = "game_play";
    BUTTON_TYPE["LOG_IN"] = "account_link";
    BUTTON_TYPE["LOG_OUT"] = "account_unlink";
    BUTTON_TYPE["POSTBACK"] = "postback";
    BUTTON_TYPE["SHARE"] = "element_share";
    BUTTON_TYPE["URL"] = "web_url";
})(BUTTON_TYPE = exports.BUTTON_TYPE || (exports.BUTTON_TYPE = {}));
var MEDIA_TYPE;
(function (MEDIA_TYPE) {
    MEDIA_TYPE["IMAGE"] = "image";
    MEDIA_TYPE["VIDEO"] = "video";
})(MEDIA_TYPE = exports.MEDIA_TYPE || (exports.MEDIA_TYPE = {}));
var AIRLINE_TRAVEL_CLASS;
(function (AIRLINE_TRAVEL_CLASS) {
    AIRLINE_TRAVEL_CLASS["ECONOMY"] = "economy";
    AIRLINE_TRAVEL_CLASS["BUSINESS"] = "business";
    AIRLINE_TRAVEL_CLASS["FIRST_CLASS"] = "first_class";
})(AIRLINE_TRAVEL_CLASS = exports.AIRLINE_TRAVEL_CLASS || (exports.AIRLINE_TRAVEL_CLASS = {}));
var REFERER_SOURCE;
(function (REFERER_SOURCE) {
    REFERER_SOURCE["MESSENGER_CODE"] = "MESSENGER_CODE";
    REFERER_SOURCE["DISCOVER_TAB"] = "DISCOVER_TAB";
    REFERER_SOURCE["ADS"] = "ADS";
    REFERER_SOURCE["SHORTLINK"] = "SHORTLINK";
    REFERER_SOURCE["CUSTOMER_CHAT_PLUGIN"] = "CUSTOMER_CHAT_PLUGIN";
})(REFERER_SOURCE = exports.REFERER_SOURCE || (exports.REFERER_SOURCE = {}));
