// fb-messenger-bot-api/dist/page/FacebookPageAPIClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Utils_1 = require("../util/Utils");
class FacebookPageAPIClient {
    /**
     * @param pageId
     * @param {string} token - Facebook FacebookPageAPIClient Token
     * @param {ProxyData} proxyData - Proxy information if behind proxy
     */
    constructor(pageId, token, proxyData) {
        this.pageId = pageId;
        this.requestData = { token };
        this.requestData = Utils_1.Utils.getProxyData(this.requestData, proxyData);
    }
    /**
     * Sets the url of the image to be added to the page.
     *
     * @param  imageUrl - The url of the image
     * @return {this}
     */
    setImageUrl(imageUrl) {
        this.url = imageUrl;
        return this;
    }
    /**
     * Sets the caption for the image to be added.
     *
     * @param imageCaption - The caption text of the image
     * @return {this}
     */
    setImageCaption(imageCaption) {
        this.caption = imageCaption;
        return this;
    }
    /**
     * Sets the text for post to be added to the page.
     *
     * @param text
     * @return {this}
     */
    setPostText(text) {
        this.message = text;
        return this;
    }
    /**
     * Sets the link of the post to be added to the page.
     * @param postLink - The url to be added to a post on the page
     * @return {this}
     */
    setPostLink(postLink) {
        this.link = postLink;
        return this;
    }
    /**
     * Adds image to Facebook FacebookPageAPIClient.
     *
     * @param {Function} cb - Optional callback for result. If no callback provided will return promise
     * @return
     */
    sendImage(cb) {
        const options = Utils_1.Utils.getRequestOptions();
        options.url += `${this.pageId}/photos`;
        options.method = 'POST';
        options.json = { caption: this.caption, url: this.url };
        return Utils_1.Utils.sendMessage(options, this.requestData, cb);
    }
    /**
     * Adds posts to Facebook FacebookPageAPIClient.
     * @param {Function} cb - Optional callback for the result. If no callback provided will return promise
     * @return
     */
    sendPost(cb) {
        const options = Utils_1.Utils.getRequestOptions();
        options.url += `${this.pageId}/feed`;
        options.method = 'POST';
        options.json = { message: this.message, link: this.link };
        return Utils_1.Utils.sendMessage(options, this.requestData, cb);
    }
}
exports.FacebookPageAPIClient = FacebookPageAPIClient;
